pkg/controller/kieapp/defaults/defaults.go

```diff
@@ -404,26 +404,15 @@ func getServersConfig(cr *api.KieApp) ([]api.ServerTemplate, error) {
 	}
 	product := GetProduct(cr.Status.Applied.Environment)
 	usedNames := map[string]bool{}
-	unsetNames := 0
 	cr.Status.Applied.Objects.Servers = serverSortBlanks(cr.Status.Applied.Objects.Servers)
 	serverSlice := cr.Status.Applied.Objects.Servers
 	for index := range serverSlice {
 		serverSet := &serverSlice[index]
 		if serverSet.Deployments == nil {
 			serverSet.Deployments = Pint(constants.DefaultKieDeployments)
 		}
-		if serverSet.Name == "" {
-			for i := 0; i < len(serverSlice); i++ {
-				serverSetName := getKieSetName(cr.Status.Applied.CommonConfig.ApplicationName, serverSet.Name, unsetNames)
-				if !usedNames[serverSetName] {
-					serverSet.Name = serverSetName
-					break
-				}
-				unsetNames++
-			}
-		}
```

Comment:
@ruromero one more change since your review...  I've decided to move this code to SetDefaults() and remove from here as redundant.